core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

```diff
@@ -0,0 +1,873 @@
+
+      translateService.onLangChange.emit();
+
+      expect(component.setPageTitle).toHaveBeenCalled();
+    }));
+
+    it('should obtain translated page title and set it', () => {
+      spyOn(translateService, 'instant').and.callThrough();
+      spyOn(pageTitleService, 'setDocumentTitle');
+
+      component.setPageTitle();
+
+      expect(translateService.instant).toHaveBeenCalledWith(
+        'I18N_LEARNER_DASHBOARD_PAGE_TITLE');
+      expect(pageTitleService.setDocumentTitle).toHaveBeenCalledWith(
+        'I18N_LEARNER_DASHBOARD_PAGE_TITLE');
+    });
+
+    it('should get static image url', () => {
+      let imagePath = '/path/to/image.png';
+
+      expect(component.getStaticImageUrl(imagePath)).toBe(
+        '/assets/images/path/to/image.png');
+    });
+
+    it('should get user profile image png data url correctly', () => {
+      expect(component.getauthorPicturePngDataUrl('username')).toBe(
+        'profile-image-url-png');
+    });
+
+    it('should get user profile image webp data url correctly', () => {
+      expect(component.getauthorPictureWebpDataUrl('username')).toBe(
+        'profile-image-url-webp');
+    });
+
+    it('should set activeSection', () => {
+      const newActiveSectionName = 'I18N_LEARNER_DASHBOARD_HOME_SECTION';
+
+      component.setActiveSection(newActiveSectionName);
+
+      expect(component.activeSection).toEqual(newActiveSectionName);
+    });
+    it('should show username popover based on its length', () => {
+      expect(component.showUsernamePopover('abcdefghijk')).toBe('mouseenter');
+      expect(component.showUsernamePopover('abc')).toBe('none');
+    });
+
+    it('should show new and old content when opening suggestion modal',
+      () => {
+        spyOn(suggestionModalForLearnerDashboardService, 'showSuggestionModal')
+          .and.returnValue(null);
+
+        let newContent = 'New content';
+        let oldContent = 'Old content';
+        let description = 'Description';
+        component.showSuggestionModal(newContent, oldContent, description);
+
+        expect(suggestionModalForLearnerDashboardService.showSuggestionModal)
+          .toHaveBeenCalledWith('edit_exploration_state_content', {
+            newContent: newContent,
+            oldContent: oldContent,
+            description: description
+          });
+      });
+
+    it('should get css classes based on status', () => {
+      expect(component.getLabelClass('open')).toBe('badge badge-info');
+      expect(component.getLabelClass('compliment')).toBe('badge badge-success');
+      expect(component.getLabelClass('another')).toBe('badge badge-secondary');
+    });
+
+    it('should get human readable status from provided status', () => {
+      expect(component.getHumanReadableStatus('open')).toBe('Open');
+      expect(component.getHumanReadableStatus('compliment')).toBe('Compliment');
+      expect(component.getHumanReadableStatus('not_actionable')).toBe(
```

Comment:
Looks like these are unused code. Can we remove the unused code instead?

Reply:
Removed as unused